lib/pyspark_preprocesing.py

```diff
@@ -149,3 +145,56 @@ def count_vectorizer(
         result = result.drop(column)
 
     return result
+
+
+def convert_types(df: DataFrame, columns: List[str], type: str) -> DataFrame:
+    """Funkcja konwertuje kolumny na określony typ danych.
+
+    Args:
+        df (DataFrame):         sparkowy DataFrame
+        columns (List[str]):    lista kolumn do przekonwerotowania
+        type (str):             typ, na który chcemy przekonwertować dane
+    Returns:
+        df: sparkowy DataFrame ze zmienionymi typami kolumn
+    """
+    assert all(x in df.columns for x in columns)
+
+    for c in columns:
+        df = df.withColumn(c, col(c).cast(type).alias(c))
+
+    return df
+
+
+def value_overwrite(
+    df: DataFrame, columns: List[str], values: List[str], category: List[str]
+) -> DataFrame:
+    """Funkcja nadpisuje wskazane kolumny poprzez wskazane wartości (lub kolumny)
+    dla okreslonych kategorii filmowych.
+
+    Args:
+        df (DataFrame):         sparkowy DataFrame
+        columns (List[str]):    lista kolumn do nadpisania
+        values (List[str]):     odpowiednio wartości (lub kolumny) jakimi chcemy nadpisać
+        category (List[str]):   kategorie dla jakich mamy nadpisywać wartości
+
+    Returns:
+        df: sparkowy DataFrame z nadpisanymi wartościami
+    """
+    assert all(x in df.columns for x in columns)
+    assert "rodzaj_produkcji" in df.columns
+    assert len(columns) == len(values)
+
+    for col, value in zip(columns, values):
+        if isinstance(value, str):
```

Comment:
dodałbym if isinstance(value, str) and value in df.columns:

Reply:
Gotowe